zenoh/src/net/routing/interceptor/access_control.rs

```diff
@@ -112,11 +109,25 @@ impl InterceptorFactoryTrait for AclEnforcer {
                     Ok(links) => {
                         for link in links {
                             for face in link.interfaces {
-                                let subject = &Subject::Interface(face.clone());
-                                if let Some(val) = self.enforcer.subject_map.get(subject) {
-                                    authn_ids.push(AuthSubject {
-                                        id: *val,
-                                        name: face,
+                                // combining each interface with remaining AuthId values to get existing combinations
+                                // NOTE: current ACL logic does not apply correctly when multiple interfaces are returned
+                                let face = Subject::Interface(face.clone());
+                                let mut subject_query = subjects.clone();
+                                subject_query.push(face);
+                                for subject_combination in
+                                    self.enforcer.subject_map.get(&subject_query)
+                                {
+                                    authn_ids.insert(AuthSubject {
+                                        id: *subject_combination.1,
+                                        name: format!(
```

Comment:
A quick "Search for references" using r-a shows that this field is only used in `tracing` events. While this path is not hot I believe that `name` ought to just be in the `Debug` or `Display` impl.